### PR TITLE
feat(eventstore): sdk

### PR DESCRIPTION
**Pull request by adlerhurst**

### CODE REVIEW

File: internal/eventstore/sdk/sdk_test.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+					},
+				},
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := Push(context.Background(), tt.args.push, tt.args.appender, tt.args.aggregaters...); (err != nil) != tt.wantErr {
+				t.Errorf("Push() error = %v, wantErr %v", err, tt.wantErr)
```
Comment (fgerschwiler) [inline internal/eventstore/sdk/sdk_test.go:181]: maybe check if its the expected error you get?

File: internal/eventstore/sdk/sdk_test.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+					return nil
+				},
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := Filter(context.Background(), tt.args.filter, tt.args.appender, nil); (err != nil) != tt.wantErr {
+				t.Errorf("Filter() error = %v, wantErr %v", err, tt.wantErr)
```
Comment (fgerschwiler) [inline internal/eventstore/sdk/sdk_test.go:69]: Check if its expected err

File: internal/project/repository/eventsourcing/model.go [hunk 3]
```diff
@@ -15,6 +52,14 @@ func ProjectFromEvents(project *Project, events ...*es_models.Event) (*Project,
...
 }
 
+func (p *Project) Changes(changed *Project) map[string]interface{} {
+	changes := make(map[string]interface{}, 1)
+	if changed.Name != "" && p.Name != changed.Name {
+		changes["name"] = changed.Name
+	}
+	return changes
+}
+
```
Comment (fgerschwiler) [inline internal/project/repository/eventsourcing/model.go:62]: Move test functions also to model_test file

File: internal/project/repository/eventsourcing/project_test.go [hunk 4]
```diff
@@ -86,7 +87,9 @@ func TestProjectByIDQuery(t *testing.T) {
 			res: res{
 				filterLen: 3,
 				wantErr:   true,
-				errFunc:   caos_errs.IsPreconditionFailed,
```
Comment (fgerschwiler) [inline internal/project/repository/eventsourcing/project_test.go:89]: I think this went wrong on merge

File: internal/project/repository/eventsourcing/project_test.go [hunk 5]
```diff
@@ -225,13 +228,15 @@ func TestProjectCreateAggregate(t *testing.T) {
 				eventLen:  1,
 				eventType: model.ProjectAdded,
 				wantErr:   true,
-				errFunc:   caos_errs.IsPreconditionFailed,
+				errFunc: func(err error) bool {
+					return caos_errs.IsPreconditionFailed(err)
+				},
```
Comment (fgerschwiler) [inline internal/project/repository/eventsourcing/project_test.go:233]: i think went wrong on merging

File: internal/project/repository/eventsourcing/project_test.go [hunk 6]
```diff
@@ -291,7 +296,9 @@ func TestProjectUpdateAggregate(t *testing.T) {
 				eventLen:  1,
 				eventType: model.ProjectChanged,
 				wantErr:   true,
-				errFunc:   caos_errs.IsPreconditionFailed,
+				errFunc: func(err error) bool {
+					return caos_errs.IsPreconditionFailed(err)
+				},
```
Comment (fgerschwiler) [inline internal/project/repository/eventsourcing/project_test.go:301]: went wrong on merging

File: internal/project/repository/eventsourcing/project_test.go [hunk 7]
```diff
@@ -306,13 +313,15 @@ func TestProjectUpdateAggregate(t *testing.T) {
...
```
Comment (fgerschwiler) [inline internal/project/repository/eventsourcing/project_test.go:318]: went wrong on merging

File: internal/project/repository/eventsourcing/project_test.go [hunk 8]
```diff
@@ -370,13 +379,15 @@ func TestProjectDeactivateAggregate(t *testing.T) {
 				eventLen:  1,
 				eventType: model.ProjectDeactivated,
 				wantErr:   true,
-				errFunc:   caos_errs.IsPreconditionFailed,
+				errFunc: func(err error) bool {
+					return caos_errs.IsPreconditionFailed(err)
+				},
```
Comment (fgerschwiler) [inline internal/project/repository/eventsourcing/project_test.go:384]: went wrong on merging

File: internal/project/repository/eventsourcing/project_test.go [hunk 9]
```diff
@@ -431,13 +442,15 @@ func TestProjectReactivateAggregate(t *testing.T) {
 				eventLen:  1,
 				eventType: model.ProjectReactivated,
 				wantErr:   true,
-				errFunc:   caos_errs.IsPreconditionFailed,
+				errFunc: func(err error) bool {
+					return caos_errs.IsPreconditionFailed(err)
+				},
```
Comment (fgerschwiler) [inline internal/project/repository/eventsourcing/project_test.go:447]: went wrong on merging

**Comment by fforootd** [comment]
it would be great to merge the PR https://github.com/caos/zitadel/pull/38 first, because this fixes a small problem with the docker tag

**Pull request merged**